doc/tutorials/dune-package-management/oxcaml.md

```diff
@@ -0,0 +1,116 @@
+# Setting up an OxCaml project
+
+As an example of how to add custom repositories and pin dependencies in projects
+using Dune Package Management, let's look at setting up a basic OxCaml project.
+[OxCaml](https://oxcaml.org/) is a compiler branch with fast moving set of
+extensions to OCaml. Some tweaks are required to set up a project that compiles
+with OxCaml when using dune package management.
+
+## Setting up the `dune-workspace`
+
+OxCaml has a custom `opam-repository` that provides compatible dependencies.
+This is specified in the `dune-workspace`.
+
+```{code-block} dune
+(lang dune 3.20)
+
+(pkg enabled)
+
+(repository
+ (name oxcaml)
+ (url git+https://github.com/oxcaml/opam-repository))
+
+(lock_dir
+ (repositories overlay oxcaml upstream))
+```
+
+In the `dune-project`, we also need to add a constraint to depend on the OxCaml
+compiler. Otherwise, the solver may pick up the latest upstream compiler, which
+does not support the OxCaml extensions.
+
+```{code-block} dune
+(lang dune 3.20)
+
+(package
+ (name hello-oxcaml)
+ (depends
+  (ocaml-variants
+   (= 5.2.0+ox))))
+```
+
+Let's add a test program that uses OxCaml syntax.
+
+::::{dropdown} `main.ml`
+:icon: file-code
+
+:::{literalinclude} oxcaml/main.ml
+:language: dune
+:::
+
+::::
+
+And to build it, we add a `dune` file.
+
+```{code-block} dune
+(executable
+ (public_name main))
+```
+
+## Building the project
+
+As usual, let's build the project with:
```

Comment:
Let's not assume users are used to the usual way of doing things here. Let's instead say, we can now build and execute our binary compiled with `OxCaml`. We use the `dune exec` command which will first build all the packages we need, including the OxCaml compiler and then builds our binary and runs it.

You should probably mention the passage of time here, since this step takes a while and dune atm doesn't give too much feedback.